sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPermissionProperties.java

```diff
@@ -45,25 +57,102 @@ public CosmosPermissionProperties setId(String id) {
      */
     CosmosPermissionProperties(String jsonString) {
         this.permission = new Permission(jsonString);
+        this.permissionName = permission.getId();
+        this.permissionMode = permission.getPermissionMode();
+        this.resourcePartitionKey = permission.getResourcePartitionKey();
+        this.resourceToken = permission.getToken();
+
+        String[] parts = Utils.trimBeginningAndEndingSlashes(permission.getResourceLink()).split("/");
+
+        if (parts.length < 4) {
+            throw new IllegalArgumentException("jsonString");
+        }
+
+        this.databaseName = parts[1];
+        this.containerName = parts[3];
+
+        if (parts.length >= 6) {
+            this.resourceName = parts[5];
+
+            if (Paths.DOCUMENTS_PATH_SEGMENT.equalsIgnoreCase(parts[4])) {
+                this.resourceKind = CosmosContainerChildResourceKind.ITEM;
+            } else if (Paths.STORED_PROCEDURES_PATH_SEGMENT.equalsIgnoreCase(parts[4])) {
+                this.resourceKind = CosmosContainerChildResourceKind.STORED_PROCEDURE;
+            } else if (Paths.USER_DEFINED_FUNCTIONS_PATH_SEGMENT.equalsIgnoreCase(parts[4])) {
+                this.resourceKind = CosmosContainerChildResourceKind.USER_DEFINED_FUNCTION;
+            } else if (Paths.TRIGGERS_PATH_SEGMENT.equalsIgnoreCase(parts[4])) {
+                this.resourceKind = CosmosContainerChildResourceKind.TRIGGER;
+            } else {
+                throw new IllegalArgumentException("jsonString");
```

Comment:
I think this is intended to have jsonString not "jsonString" ?

Reply:
"jsonString" is the name of the argument, not the content...